docs/contributing/coding-guidelines/ros-nodes/console-logging.md

```diff
@@ -1,5 +1,128 @@
 # Console logging
 
-!!! warning
+ROS 2 logging is a powerful tool for understanding and debugging ros node.
 
-    Under Construction
+The logging subsystem in ROS 2 delivers logging messages to a variety of targets, including console, disk and ROS 2 network.
+
+This page focus on console logging and shows some useful example in practice. for detailed information, please see [logging documentation](https://docs.ros.org/en/humble/Concepts/About-Logging.html).
+
+## API
+
+There are four pattern of logging function:
+
+```bash
+RCLCPP_{severity}
+RCLCPP_{severity}_STREAM
+RCLCPP_{severity}_{condition}
+RCLCPP_{severity}_STREAM_{condition}
+```
+
+`STREAM` means output message in api is C++stream-style else printf style.
+
+`severity` have five choice:
+
+- `DEBUG, INFO, WARN, ERROR or FATAL`
+
+Set severity level to `DEBUG` only outputs message in debug mode.
+
+`condition` choices are:
+
+- `THROTTLE` - output the given printf-style message no more than the given rate
+- `ONCE` - output the given printf-style message only the first time this line is hit
+- `EXPRESSION` - output the given printf-style message only if the given expression is true
+- `FUNCTION` - output the given printf-style message only if the given function returns true
+- `SKIPFIRST` - output the given printf-style message all but the first time this line is hit
```

Comment:
I feel this part can be delegated to the ROS 2 Documentation.
@alanmengg What is the reason that you want to write it here?

```suggestion
```